### PR TITLE
feat: update reference and task models

### PR DESCRIPTION
**Pull request by ryanfang5**
Created TaskNested which is used in the Reference model.

Also allow None values for certain fields in the Reference model.

### CODE REVIEW

File: virtool_core/models/reference.py [hunk 1]
```diff
@@ -83,13 +83,13 @@ class ReferenceMinimal(AnalysisReference):
     data_type: ReferenceDataType
     groups: List[ReferenceUser]
     installed: ReferenceInstalled = None
-    internal_control: str
-    latest_build: ReferenceBuild
+    internal_control: str = None
+    latest_build: ReferenceBuild = None
```
Comment (igboyes) [inline virtool_core/models/reference.py:87]: We should not use default values in models unless totally necessary. It makes things less explicit and well-defined.
```suggestion
    latest_build: Optional[ReferenceBuild]
```

File: virtool_core/models/reference.py [hunk 2]
```diff
@@ -83,13 +83,13 @@ class ReferenceMinimal(AnalysisReference):
     data_type: ReferenceDataType
     groups: List[ReferenceUser]
     installed: ReferenceInstalled = None
-    internal_control: str
-    latest_build: ReferenceBuild
+    internal_control: str = None
```
Comment (igboyes) [inline virtool_core/models/reference.py:86]: We should not use default values in models unless totally necessary. It makes things less explicit and well-defined.
```suggestion
    internal_control: Optional[str]
```

File: virtool_core/models/reference.py [hunk 3]
```diff
@@ -19,9 +19,9 @@ class ReferenceDataType(str, enum.Enum):
 
 
 class ReferenceUser(UserNested):
-    count: int
+    count: Optional[int]
     build: bool
-    created_at: datetime
+    created_at: Optional[datetime]
```
Comment (igboyes) [inline virtool_core/models/reference.py:24]: Looks like this is not returned from the API so it should just be removed from the model.
```suggestion
```

File: virtool_core/models/reference.py [hunk 4]
```diff
@@ -19,9 +19,9 @@ class ReferenceDataType(str, enum.Enum):
 
 
 class ReferenceUser(UserNested):
-    count: int
+    count: Optional[int]
```
Comment (igboyes) [inline virtool_core/models/reference.py:22]: Not sure where this came from. Remove.
```suggestion
```

**Review by igboyes — CHANGES_REQUESTED** [review]

**Comment by ryanfang5** [comment]
Made adjustments to the ReferenceUser model which are more accurate to the api responses.

File: virtool_core/models/reference.py [hunk 5]
```diff
@@ -18,13 +18,26 @@ class ReferenceDataType(str, enum.Enum):
...
+class ReferenceGroupOrUser(ReferenceRights):
+    id: str
+
+
+class ReferenceUsers(ReferenceGroupOrUser):
+    build: bool
+    handle: str
+
+
+class ReferenceContributors(UserNested):
```
Comment (igboyes) [inline virtool_core/models/reference.py:37]:
```suggestion
class ReferenceContributor(UserNested):
```

File: virtool_core/models/reference.py [hunk 6]
```diff
@@ -18,13 +18,26 @@ class ReferenceDataType(str, enum.Enum):
...
 
-class ReferenceUser(UserNested):
+class ReferenceRights(BaseModel):
     build: bool
     modify: bool
     modify_otu: bool
     remove: bool
 
 
+class ReferenceGroupOrUser(ReferenceRights):
```
Comment (igboyes) [inline virtool_core/models/reference.py:28]: Please split into separate models the inherit from `ReferenceRights`. If they diverge at some point (they will), a refactor less painful.
```suggestion
class ReferenceGroupOrUser(ReferenceRights):
```

**Review by igboyes — CHANGES_REQUESTED** [review]

**Review by igboyes — APPROVED** [review]

**Pull request merged**